games/guess-it/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+        };
+        unescape(serde_json::to_string(&serializable).unwrap_or("".to_string())
+            .as_mut_str())
+            .unwrap_or("".to_string())
+    }
+}
+
+#[derive(TypeId, Decode, Encode, Clone, Deserialize, Serialize)]
+pub struct GameSerialized {
+    pub name: String,
+    pub state: State,
+    pub bet: String,
+    pub players: HashMap<String, Player>,
+    pub winner: String,
+    pub last_roll: String,
+}
+impl fmt::Debug for GameSerialized {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let formatted = serde_json::to_string(&self).unwrap().replace("\\", "");
+        write!(f, "{}", formatted)
+    }
+}
+impl GameSerialized {
+    pub fn empty(players: Option<HashMap<String, Player>>) -> Self {
+        Self {
+            name: "".to_string(),
+            state: State::AcceptingChallenger,
+            bet: 0.to_string(),
+            players: players.unwrap_or(HashMap::new()),
+            winner: NonFungibleKey::from(0).to_string(),
+            last_roll: 0.to_string(),
+        }
+    }
+}
+
+#[derive(Debug, TypeId, Describe, Decode, Encode, Clone, Serialize, Deserialize)]
+pub struct Player {
+    guess: String,
+}
+impl Player {
+    pub fn empty() -> Player {
+        return Self {
+            guess: "".into(),
+        }
+    }
+
+    pub fn get_guess(&self) -> String { self.guess.clone() }
+    pub fn make_guess(&mut self, guess: String) { self.guess = guess; }
+}
+
+#[derive(TypeId, Describe, Decode, Encode)]
+#[derive(NonFungibleData)]
+struct PlayerNft {
+    name: String,
+}
+
+blueprint! {
+    struct GuessIt {
+        bank: Vault,
+        badges: Vault,
+        badge_ref: ResourceDef,
+        game: Game,
+    }
+
+    impl GuessIt {
+        pub fn create(name: String, bet: String) -> Component {
+            let player_badges: Bucket = ResourceBuilder::new_non_fungible()
+                .metadata("name", "Guess-It Game")
+                .metadata("symbol", "GIG")
+                .initial_supply_non_fungible([
+                    (
+                        NonFungibleKey::from(Uuid::generate()),
+                        PlayerNft {name: "Player 1".to_string()},
+                    ),
+                    (
+                        NonFungibleKey::from(Uuid::generate()),
+                        PlayerNft {name: "Player 2".to_string()},
+                    ),
+                ]);
+            let player_badge_def = player_badges.resource_def();
+
+            Self {
+                bank: Vault::new(RADIX_TOKEN),
+                badges: Vault::with_bucket(player_badges),
+                badge_ref: player_badge_def,
+                game: Game::new(None, name, bet),
+            }.instantiate()
+        }
+
+        pub fn join(&mut self, mut payment: Bucket) -> (Bucket, Bucket, String) {
+            assert_eq!(self.game.state, State::AcceptingChallenger, "Not accepting challengers!");
+            assert!(self.game.players.len() <= 2, "Too many players!");
```

Comment:
Should be `self.game.players.len() < 2` ?

Reply:
I am under the impression that <2 would only work for 0-1 players. 
the assert is attempting to limit to 2 players and error "Too many players!" if any more try to join

Reply:
There was some confusion about the number of players admitted into the game, so the README.md was updated to reflect that only 2 players are allowed to enter.